### PR TITLE
Avoid non-standard use of `uint`/`ushort`

### PR DESCRIPTION
**Pull request by donn**
* Change uses of `uint` to `odb::uint` where available (by explicit namespace access or `using` as appropriate) or by expansion to `unsigned int` in files where `odb` is not imported
* Change the few uses of `ushort` to `uint16_t`, importing `cstdint` as appropriate
* `src/dbSta/src/dbSta.i`: Change import of `Exception.i` to match that of other `.i` files (Current import only works when `USE_SYSTEM_OPENSTA` is `OFF`)
* Added a recommended compile environment to `./etc/shell.nix` for posterity

---

Granted, this only helps on some freak setups such GCC-based compile environments on aarch64 macOS. I can confirm it doesn't hurt at least (and I hope the CI backs me up on that. 🙂)

### CODE REVIEW

File: src/drt/src/serialization.h [hunk 1]
```diff
@@ -56,6 +56,9 @@
 #include "global.h"
 #include "odb/dbTypes.h"
 #include "odb/geom.h"
+
+using uint = odb::uint;
```
Comment (maliberty) [inline src/drt/src/serialization.h:60]: You putting this back into the global namespace.

Comment (donn) [reply src/drt/src/serialization.h:60]: Good catch- will handle

File: src/drt/src/db/drObj/drNet.h [hunk 2]
```diff
@@ -104,14 +104,14 @@ class drNet : public drBlockObject
...
   bool isRouted() const { return routed_; }
   const std::vector<frRect>& getOrigGuides() const { return origGuides_; }
-  ushort getPriority() const { return priority_; }
+  uint16_t getPriority() const { return priority_; }
   // setters
   void incPriority()
   {
-    if (priority_ < std::numeric_limits<ushort>::max())
+    if (priority_ < std::numeric_limits<uint16_t>::max())
       priority_++;
```
[the remark this answers is not in the document]
Comment (donn) [reply src/drt/src/db/drObj/drNet.h:112]: Not that I disagree but this technically wasn't part of my changes

[the remark this answers is not in the document]
Comment (maliberty) [reply src/drt/src/db/drObj/drNet.h:112]: Once you touch the line clang-tidy will report on it.  If you want to leave it out that's ok

[the remark this answers is not in the document]
Comment (donn) [reply src/drt/src/db/drObj/drNet.h:112]: yeah i'd rather not. likely will end up tripping up clang-format in return

**Review by maliberty — APPROVED** [review]

**Pull request merged**